pkg/cluster/kubernetes/sync.go

```diff
@@ -215,8 +215,16 @@ func (c *Cluster) getAllowedResourcesBySelector(selector string) (map[string]*ku
 	resources := []*meta_v1.APIResourceList{}
 	for i := range sgs.Groups {
 		gv := sgs.Groups[i].PreferredVersion.GroupVersion
-		// exclude the *.metrics.k8s.io resources to avoid querying the cluster metrics
-		if sgs.Groups[i].Name != "metrics.k8s.io" && !strings.HasSuffix(sgs.Groups[i].Name, ".metrics.k8s.io") {
+
+		excluded := false
+		for _, exg := range c.groupExcludeList {
+			if strings.HasSuffix(sgs.Groups[i].Name, exg) {
```

Comment:
Wonder if suffix matches are sufficient for all use-cases, or that the next enhancement request is going to be glob matches.

Reply:
For groups I think suffix matches are covering all use-cases but if we want to exclude Kubernetes Kinds then glob should be used.

Reply:
I had made a more flexible design a while ago, see https://github.com/fluxcd/flux/issues/2642

Yeah, I think we should use globing instead.

Reply:
That issue refers to kinds not groups, this is a different feature from what you are describing in the issue. I see no reason why would you need blogling when groups follow the DNS format, if you want to exclude all certmanager's groups you would add `certmanager.k8s.io`.

Reply:
Yes, it contains all groups versions and kinds, it's a superset of this PR. I think we should implement that since it's more flexible and the approach in this PR is not extensible (making it hard to add kinds later on).

I am happy to implement it myself if you don't feel like updating the PR.

Reply:
Ok I've changed to traget resources with glob expressions.

Reply:
Only GroupVersions are targeted. So we should either include Kinds (and not only use it for discovery) or rename the argument.

Reply:
Actually, I am on the fence about this. `--k8s-exclude-resources` is probably fine.